### ISSUE TITLE
Display consecutive successes in the same way as fails

### ISSUE THREAD

**Issue by bkosty** [jan-molak/jenkins-build-monitor-plugin#247]
I just installed Build Monitor and am loving it.  I don't know if this is a feature request or a setting I can't seem to find.   Currently, if there are a string of fails it will show "X builds have failed"  Is there a way to show successes in the same way?

**Comment by jan-molak** [comment]
Hey @bkosty and thanks for your feedback, glad that you found it useful :-)
There's no such option at present, but I like the idea very much! It could act akin to [Seinfeld's Strategy](http://lifehacker.com/281626/jerry-seinfelds-productivity-secret) and add a positive reinforcement twist to Build Monitor 😃 

I'll add it to the backlog and will try to get it implemened when I have a moment. Of course pull requests are also welcome!

**Comment by vanta** [comment]
It would be nice to see a number of days that build is green for. 

Teams can compete to have longest green builds strikes. 

Btw. I cannot find a plugin for Jenkins to get such information...

**Comment by jan-molak** [comment]
Yeah, @vanta, that's what I was thinking :-) Theoretically, each build contains information about when it was executed, so the day could be derived from that. However:
- calculating this information on every call to the Build Monitor API could be expensive and make it slow
- since disk usage is often a significant constraint on Jenkins servers, people might choose to only keep a handful of recent builds around (ie: last two days worth).

Given the above, the information about the length of the strike might need to be calculated every now and then and stored _somewhere_ ...

Thoughts?
J

**Comment by vanta** [comment]
@jan-molak I don't know the Jenkins internals, but I'd see it this way, that every failed build reset the date and counter and every successful builds increments counter. Then, you can get this info from latest build (or job in general?). Disk usage should be pretty small as you only need to store counter and date per job, not per every build. Even per build it's small fingerprint (few bytes?).

**Comment by jan-molak** [comment]
Sure, agreed, I was talking about the disk usage in relation to [storing the builds themselves](https://wiki.jenkins-ci.org/display/JENKINS/Discard+Old+Build+plugin), which could take hundreds of megabytes on some projects. Storing cached metrics would be the way to go, but there isn't such mechanism available within Build Monitor just yet - ie. someone would need to write it :)
